web/modules/custom/hoeringsportal_audit_log/src/Routing/RoutingChangeListener.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace Drupal\hoeringsportal_audit_log\Routing;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\os2web_audit\Service\Logger;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Symfony\Component\HttpKernel\Event\ControllerEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Drupal\hoeringsportal_audit_log\Form\SettingsForm;
+use Drupal\node\Entity\Node;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Route change listener.
+ */
+final class RoutingChangeListener implements EventSubscriberInterface {
```

Comment:
It's always hard to name things, but “routing change” does not capture the essence of what this listener actually does. “controller listerener” may be a better name and reflect what is actually done.